app/models/team.rb

```diff
@@ -1,3 +1,10 @@
 class Team < ApplicationRecord
     validates :name, presence: true
+    has_many :players
+    has_many :matches
+    has_many :player_stats, through: :matches
+    has_many :stats, through: :player_stats
+    # not sure about the above three lines. in fact, the matches relationship for teams doesn't work,
+    # nor does the player_stats or stats of course. it seems to look for a team_id column in matches model when i 
+    # call Team.first.matches, which doesn't exist (only home_team_id and away_team_id exist). Not sure how to fix
```

Comment:
that's right - you probably have to specify the key something like - `has_many: :matches, foreign_key: :home_team_id`

but since we have both the `home_key_id` and the `away_team_id` and we want to include both, it's probably better to define a method something like - 
```
def matches
  Match.where("home_team_id = ? or away_team_id = ?", id, id)
end
```

Reply:
so that function says for a given team (say Chelsea), return all the matches such that: 
match's home team id = Chelsea's id or 
match's away team id = Chelsea's id?